spec/markettoken/market-token.spec.ts

```diff
@@ -0,0 +1,94 @@
+import * as ganache from 'ganache-cli'
+import Web3 from 'web3'
+import { Contract } from 'web3/types'
+import MarketToken from '../../src/contracts/market-token'
+import { MARKET_TOKEN_ABI, ONE_ETHER, ONE_GWEI } from  '../../src/constants'
+import { Return } from '../../src/@types'
+import {
+  deploy,
+  readBytecode,
+  call,
+  transact,
+} from '../../src/helpers'
+
+const provider:any = ganache.provider(),
+  w3 = new Web3(provider, undefined, {defaultBlock: 'latest',
+    transactionConfirmationBlocks: 1, transactionBlockTimeout: 5}),
+    toBN = w3.utils.toBN
+
+let token:MarketToken,
+  accounts:string[],
+  deployed:Contract
+
+describe('Market Token', () => {
+  beforeAll(async () => {
+    accounts = await w3.eth.getAccounts()
+
+    // deploy it...
+    const bin:string = readBytecode('markettoken')
+
+    deployed = await deploy(w3, accounts[0], MARKET_TOKEN_ABI,
+      bin, [accounts[0], ONE_ETHER])
+
+    // now we can instantiate the HOC
+    token = new MarketToken(accounts[0])
+    await token.at(w3, deployed.options.address)
+  })
+
+  describe('Class methods for Market Token', () => {
+
+    it('calls setPrivileged correctly', async () => {
+      const defaults = await token.setPrivileged("listing", "reserve", {})
+      let tx = defaults[0]
+      let opts = defaults[1]
+      expect(tx).not.toBeNull(accounts[0])
+      expect(opts).not.toBeNull()
+      expect(Object.keys(opts).indexOf('gas')).toBeGreaterThan(-1)
+      expect(Object.keys(opts).indexOf('from')).toBeGreaterThan(-1)
+      expect(Object.keys(opts).indexOf('gasPrice')).toBeGreaterThan(-1)
+      expect(opts['gas']).toBeGreaterThan(0)
+      let from = opts['from']
+      expect(from).not.toBeNull()
+      expect(from!.trim().length).toBeGreaterThan(0)
+      let gasPrice = opts['gasPrice']
+      expect(gasPrice).not.toBeNull()
+      expect(parseInt(gasPrice)).toBeGreaterThan(0)
+    })
+
+    it('calls getPrivileged correctly', async () => {
+      const defaults = await token.getPrivileged({})
+      let tx = defaults[0]
+      let opts = defaults[1]
+      expect(tx).not.toBeNull(accounts[0])
+      expect(opts).not.toBeNull()
+      expect(Object.keys(opts).indexOf('gas')).toBeGreaterThan(-1)
```

Comment:
i think a simple `gas in opts` is fine or a `opts.gas >= 121000` (note on that number. on the EVM 21000 gas is the least anything can cost. We pad every gas estimate by 100k. Thus, the gas returned for something unknown or unstated would be 121k)

Reply:
updated in latest commit to test >= 121000